components/script/dom/globalscope.rs

```diff
@@ -227,6 +227,14 @@ impl GlobalScope {
         self.uncaught_rejections.borrow_mut().push(Heap::boxed(rejection.get()));
     }
 
+    pub fn remove_uncaught_rejection(&self, rejection: HandleObject) {
+        let mut uncaught_rejections = self.uncaught_rejections.borrow_mut();
+
+        if let Some(index) = uncaught_rejections.iter().position(|promise| *promise == Heap::boxed(rejection.get())) {
```

Comment:
I think we can do `**promise == rejection.get()` instead.

Reply:
After trying to use `**promise == rejection.get()`, I found it will get following error.

```rust
error[E0308]: mismatched types
   --> components/script/dom/globalscope.rs:233:89
    |
233 |         if let Some(index) = uncaught_rejections.iter().position(|promise| **promise == rejection.get()) {
    |                                                                                         ^^^^^^^^^^^^^^^ expected struct `mozjs_sys::jsgc::Heap`, found *-ptr
    |
    = note: expected type `mozjs_sys::jsgc::Heap<*mut js::jsapi::JSObject>`
               found type `*mut js::jsapi::JSObject`
```

So, I think maybe we still need to wrap the `rejection.get()` within a `Heap`?